### PR TITLE
Add std::unique_ptr<type> to Perl5.

### PR DESCRIPTION
**Pull request by erezgeva**
Signed-off-by: Erez Geva <ErezGeva2@gmail.com>

### CODE REVIEW

**Comment by ojwb** [comment]
The CI failures seem to be unrelated.

Looks plausible to me, but I'm not familiar with existing unique_ptr support so it'd be better to get an OK from someone who is.

**Review by wsfulton — APPROVED** [review]
Looks good to me. Just waiting for failures on master to be fixed before merging this.

**Pull request merged**
